### PR TITLE
Add non-panicking abs() functions to all signed integer types.

### PR DESCRIPTION
**Pull request by jethrogb**
[Rendered](https://github.com/jethrogb/rfcs/blob/wrapping_abs/text/0000-no-panic-abs.md)

### CODE REVIEW

**Comment by ExpHP** [comment]
Can you name a use case?  It's hard for me to picture when it would be useful to receive a value of a different type when you want the absolute value of a number.

An alternative could be `checked_abs` (which would use `checked_neg` for negatives).

**Comment by jethrogb** [comment]
Obtaining the absolute value is going to always result in a positive number, so why would you not want to store that in an unsigned type? In any case, one example where this is useful for converting a number to a sign-and-magnitude representation:

```
(signed.is_negative(),signed.wrapping_abs() as u64)
```

The alternative you mention is not really an alternative but it is something that could be added to this RFC.

**Comment by jethrogb** [comment]
Added `checked_abs` and `overflowing_abs` to the RFC. I didn't add `saturating_abs` since there is no `saturating_neg`.

**Comment by glaebhoerl** [comment]
Rendered link seems to be broken

**Comment by glaebhoerl** [comment]
earlier proposal for reference: [Change abs() to return unsigned integers](https://github.com/rust-lang/rfcs/pull/1017)

**Comment by jethrogb** [comment]
> Rendered link seems to be broken

Fixed

File: text/0000-no-panic-abs.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	} else {
+		self
+	}
+}
+```
+
+This RFC proposes to add the following:
+
+```rust
+pub fn checking_abs(self) -> Option<Self> {
```
Comment (brson) [inline text/0000-no-panic-abs.md]: `checked_abs`, not `checking`, would seem to be more consistent with existing conventions.

**Comment by brson** [comment]
Do all these have concrete use cases or are they just here to provide symmetry to other numeric methods? ISTM like `checked_abs` is the only one that really matters.

File: text/0000-no-panic-abs.md (hunk 1 above)
Comment (jethrogb) [inline text/0000-no-panic-abs.md]: Yes, of course.

**Comment by jethrogb** [comment]
See above for a usecase for wrapping. I don't know of one for overflowing.

**Comment by alexcrichton** [comment]
🔔 This RFC is now entering its week-long **final comment period** 🔔 

The libs team is leaning towards merging this as is, and we'd also be willing to just take a PR for this as well if you'd prefer to take that route @jethrogb :)

**Comment by jethrogb** [comment]
https://github.com/rust-lang/rust/pull/35058

**Comment by alexcrichton** [comment]
Thanks @jethrogb! In that case I'll just close this in favor of that.

**Pull request closed without merging**